src/three-components/PMREMGenerator.ts

```diff
@@ -56,6 +56,10 @@ export const generatePMREM =
           cubeLods.forEach((target) => {
             target.dispose();
           });
+          meshes.forEach((mesh) => {
+            (mesh.material as Material).dispose();
+            mesh.geometry.dispose();
```

Comment:
Disposing of the PMREM materials will be a significant hit on performance (it takes longer to compile them than to run them). If we'd like to control it, perhaps we should make it a class again that just holds the materials and is a member of TextureUtils, so that it's only disposed of when TextureUtils is disposed?

Reply:
We need to dispose of them _or_ we need to ensure they are re-used. The profile suggested that we were holding very accumulating very large buffers in memory.

Reply:
I'm fine with disposing of the geometry, but the shaders are reused. Of course you could ditch them if you never wanted to load another environment map, but we don't really have a way to reason about that right now.

Reply:
I will take a stab at caching them so that they can be re-used and properly disposed of. [As we know for our earlier spelunking](https://github.com/mrdoob/three.js/blob/master/src/renderers/webgl/WebGLPrograms.js#L293), `Material` instances need to be disposed _eventually_, otherwise their backing programs will leak.